src/jit/optimizer.cpp

```diff
@@ -7757,14 +7757,16 @@ void Compiler::optComputeLoopSideEffectsOfBlock(BasicBlock* blk)
                                     ArrayInfo arrInfo;
                                     bool      b = GetArrayInfoMap()->Lookup(addrArg, &arrInfo);
                                     assert(b);
-                                    CORINFO_CLASS_HANDLE elemType =
+                                    CORINFO_CLASS_HANDLE elemTypeEq =
```

Comment:
In all other places when calling  `EncodeElemType` we assign  the resulting "class handle" to **elemTypeEq**.  This method uses the low bit of the returned "class handle" to encode the non-struct primitive types.

I simply made thsi section consistenmt with all of the other uses and introduced a couple of temp locals.